src/btree/bt_curnext.c

```diff
@@ -188,15 +188,16 @@ __cursor_var_append_next(
 
         if (cbt->upd_value->type == WT_UPDATE_INVALID) {
             ++*skippedp;
+            cbt->non_deleted_updates = true;
```

Comment:
Nitpick: (feel free to ignore this suggestion)
I find it easier to read the code with the variable name `all_updates_deleted`. `non_deleted_updates` when set to false has two negations which can make things harder to read.

Basically, we could set `all_updates_deleted` to true, until we find something otherwise and in that case make `all_updates_deleted` false.

Reply:
I like the idea. I used the name `all_deleted_items` as the variable name. I feel it is better if we remove the `updates`.